frontend/packages/topology/src/data-transforms/DataModelProvider.tsx

```diff
@@ -19,30 +17,71 @@ interface DataModelProviderProps {
   children?: React.ReactNode;
 }
 
-export const getNamespacedDynamicModelFactories = (
-  factories: LoadedExtension<DynamicTopologyDataModelFactory>[],
-) =>
-  factories.map(({ properties, ...ext }) => {
-    return {
-      ...ext,
-      properties: {
-        ...properties,
-        resources: (namespace: string) =>
-          Object.assign(
-            {},
-            ...Object.entries(properties.resources).map(([k, v]) => {
-              const kind = v?.model?.version
-                ? referenceForExtensionModel(v.model)
-                : v?.model
-                ? referenceForModel(modelForGroupKind(v.model?.group, v.model?.kind))
-                : v?.opts?.kind;
+const WatchResourcesHookResolver: React.FC<{
+  useResources: ExtensionHook<WatchK8sResources<any>>;
+  onResolve: (resources: WatchK8sResources<any>) => void;
+}> = ({ useResources, onResolve }) => {
+  const [resources, loaded] = useResources();
+  React.useEffect(() => {
+    if (loaded) {
+      onResolve(resources);
+    }
+  }, [loaded, onResolve, resources]);
+  return null;
+};
+
+export const DynamicWatchResourcesLoader: React.FC<{
+  children: (props: WatchK8sResources<any>, loaded: boolean) => React.ReactNode;
+}> = ({ children }) => {
+  const [watchResources, setWatchResources] = React.useState<WatchK8sResources<any>>({});
+  const [modelFactories, loaded] = useResolvedExtensions<DynamicTopologyDataModelFactory>(
+    isDynamicTopologyDataModelFactory,
+  );
+  const modelResources = React.useMemo(
+    () =>
+      modelFactories
+        ?.map((m) => ({ id: m.properties.id, resourceHook: m.properties.resources }))
+        .filter(({ resourceHook }) => !!resourceHook),
+    [modelFactories],
+  );
+
+  const onResolve = React.useCallback(
+    (resources: WatchK8sResources<any>) =>
+      setWatchResources((prevResources) => ({ ...prevResources, ...resources })),
+    [],
+  );
+
+  const resourcesLoaded = React.useMemo(() => loaded && !!watchResources.length, [
```

Comment:
Shouldn't this look for resolving all the resource hooks?

Reply:
Updated to check for all resource hooks before returning true.